File: 프로그래머스/1/42862. 체육복/체육복.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    
    sort(lost.begin(), lost.end());
    sort(reserve.begin(), reserve.end());
    
    for(int i=0; i<reserve.size(); i++){
        for(int j=0; j<lost.size(); j++){
            if(reserve[i] == lost[j]) {
                lost[j] = -999;
                reserve[i] = -9999;
                break;
            }
        }
    }
    
    for(int i=0; i<reserve.size(); i++){
        for(int j=0; j<lost.size(); j++){
            if(reserve[i] == lost[j] || reserve[i]-1 == lost[j] || reserve[i]+1 == lost[j]) {
                lost[j] = -999;
                break;
            }
        }
    }
    
    for(int i=0; i<lost.size(); i++){
        if(lost[i] !=-999){
            n -= 1;
        }
    }
    
    answer = n;
    
    return answer;
}

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;
        
        ArrayList<Integer> arr = new ArrayList<>();
        
        
        for(int i=0; i<lost.length; i++){
            arr.add(lost[i]);
        }
        
        Collections.sort(arr);
        Arrays.sort(reserve);
        
        for(int i=0; i<reserve.length; i++){
            for(int j=0; j<arr.size(); j++){
                if(reserve[i] == arr.get(j)){
                    reserve[i] = -1;
                    arr.remove(arr.get(j));
                }
            }
        }
        
        for(int i=0; i<reserve.length; i++){
            for(int j=0; j<arr.size(); j++){
                if(reserve[i]-1 == arr.get(j) || reserve[i] + 1== arr.get(j)){
                    reserve[i] = -1;
                    arr.remove(arr.get(j));
                }
            }
        }
        
        int k = arr.size();
        answer = n - k;
        
        return answer;
    }
}